api/docker-entrypoint.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+
+info()
+{
+    echo "Info: $*"
+}
+
+warn()
+{
+    echo "Warning: $*"
+}
+
+fatal()
+{
+    echo "Fatal: $*"
+}
+
+get_couchdb_url()
+{
+  echo "http://$COUCHDB_USER:$COUCHDB_PASSWORD@$COUCHDB_SERVICE_NAME:5985"
+}
+
+wait_for_couchdb()
+{
+  local n=60
+
+  while [ "$n" -gt 0 ]; do
+    curl "`get_couchdb_url`" &>/dev/null
+    [ "$?" -eq 0 ] && return 0
+    sleep 3
+    n=$[$n-1]
+  done
+
+  return 1
+}
+
+is_setup_needed()
```

Comment:
I love this! That `  ! is_existing_user 'medic-api'` is a weird implementation detail and a lesser version of this code might have that elsewhere inline, where nobody would be able to easily figure out what its purpose is. Having a function with a name on it is just the thing to do. I know this is a little thing, but it's made me really happy to see it done like this. Thanks!